tests/ui/dashboard.spec.ts

```diff
@@ -0,0 +1,137 @@
+
+  await expect(page).toHaveURL(
+    `${baseURL}me/records/?q=test&l=list&p=1&s=10&sort=bestmatch`
+  );
+
+  await expect(page.getByTestId("aggregation-count")).toContainText("0");
+});
+
+test("filter draft", async ({ page, baseURL }) => {
+  await page.goto("/me/records/");
+  await page.getByTestId("filter-button").click();
+
+  await expect(page.locator(".sidebar .facet-list")).toBeVisible();
+  // get all aggregation buckets
+  const numberOfAggs = await page
+    .locator(".sidebar .ui.facet")
+    .locator("visible=true")
+    .count();
+
+  const randomAggIndex = getRandomInt(numberOfAggs);
+
+  const selectedAgg = page
+    .locator(".ui.facet")
+    .locator("visible=true")
+    .nth(randomAggIndex);
+
+  // get all checkboxes in certain bucket
+  const numberOfCheckboxes = await selectedAgg
+    .locator(".list .item .checkbox")
+    .locator("visible=true")
+    .count();
+
+  const randomCheckboxIndex = getRandomInt(numberOfCheckboxes);
+
+  const selectedCheckbox = selectedAgg
+    .locator(".list .item .checkbox input")
+    .nth(randomCheckboxIndex);
+
+  const selectedBucketCount = await selectedAgg
+    .locator(".list .item .facet-count")
+    .locator("visible=true") 
+    .nth(randomCheckboxIndex)
+    .textContent()
+    .then((text) => {
+      if (!text) throw new Error("Count content is empty");
+      const index = text.indexOf(" ");
+      return text.substring(0, index);
+    });
+
+  //click the checkbox
+  await selectedCheckbox.click({ force: true });
+
+  const isChecked = await selectedCheckbox.evaluate(
+    (element) => element.checked
+  );
+  expect(isChecked).toBeTruthy();
+
+  await expect(page.getByTestId("aggregation-count")).toContainText(
+    `${selectedBucketCount}`
+  );
+  // clear filters
+  await page
+    .locator('.tablet button[name="clear"]')
+    .locator("visible=true")
+    .click();
+  await expect(page).toHaveURL(
+    `${baseURL}me/records/?q=&l=list&p=1&s=10&sort=newest`
+  );
+});
+
+test("redirection to form", async ({ page, baseURL }) => {
+  await page.goto("/me/records/");
+  await page.locator("#invenio-burger-toggle").click();
+  const btn = page.getByTestId("newupload-button");
+
+  await btn.click({ force: true });
+  await expect(page).toHaveURL(`${baseURL}docs/_new`);
+});
+
+test("pagination", async ({ page, baseURL }) => {
+  await page.goto("/me/records/");
+  await page.locator(".pagination a").filter({ hasText: /^5$/ }).click();
+  await expect(page).toHaveURL(
+    `${baseURL}me/records/?q=&l=list&p=5&s=10&sort=newest`
+  );
+});
+
+test("get records", async ({ page, baseURL }) => {
+  await page.goto("/me/records");
+  await page.locator("#invenio-burger-toggle").click();
+  await page.getByTestId("dashboard-redir-btn").click();
+
+  await expect(page).toHaveURL(
+    `${baseURL}me/records/?q=&l=list&p=1&s=10&sort=newest`
+  );
+
+  await expect(page.getByTestId("aggregation-count")).toContainText("20");
```

Comment:
Where does that 20 records come from?

Reply:
What is this "aggregation-count" element? Is it count of items in some aggregation value? Or result count?